Development/nmos/authorization_redirect_api.cpp

```diff
@@ -0,0 +1,355 @@
+                    try
+                    {
+                        client_assertion = jwt_generator::create_client_assertion(client_id, client_id, token_endpoint, client_assertion_lifespan, rsa_private_key, keyid);
+                    }
+                    catch (const std::exception& e)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization flow token request Create Client Assertion error: " << e.what();
+                        result = details::make_authorization_flow_error_response(status_codes::InternalError, e);
+                        authorization_state.authorization_flow = authorization_state::failed;
+                    }
+
+                    if (authorization_state::failed == authorization_state.authorization_flow)
+                    {
+                        with_write_lock(model.mutex, [&]
+                        {
+                            model.notify();
+                        });
+
+                        set_reply(res, result.first, !result.second.is_null() ? result.second : nmos::make_error_response_body(result.first));
+
+                        return pplx::task_from_result(true);
+                    }
+                }
+
+                // exchange authorization code for bearer token
+                request_token = details::request_token_from_redirected_uri(client, version, req.request_uri(), response_type, client_id, client_secret, scope, redirect_uri, state, code_verifier, token_endpoint_auth_meth, client_assertion, gate);
+
+                auto request = request_token.then([&model, &authorization_state, &scope, &gate](const web::http::oauth2::experimental::oauth2_token& bearer_token)
+                {
+                    auto lock = authorization_state.write_lock();
+
+                    // signal authorization_flow that bearer token has just been received
+                    authorization_state.authorization_flow = authorization_state::access_token_received;
+
+                    // update bearer token cache, which will be used for accessing protected APIs
+                    authorization_state.bearer_token = bearer_token;
+
+                    slog::log<slog::severities::info>(gate, SLOG_FLF) << utility::us2s(bearer_token.scope()) << " bearer token updated";
+
+                }).then([&](pplx::task<void> finally)
+                {
+                    auto lock = authorization_state.write_lock(); // in order to update shared state
+
+                    try
+                    {
+                        finally.get();
+                        result = { status_codes::OK, web::json::value_of({{ U("status"), U("Bearer token received") }}, true) };
+                    }
+                    catch (const web::http::http_exception& e)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization flow token request HTTP error: " << e.what() << " [" << e.error_code() << "]";
```

Comment:
I guess we should be consistent with the name of the grant. i.e. we should either use "authorization flow" or "authorization code flow" everywhere.  RFC 8252 uses "authorization flow" so maybe we should use that everywhere?